source/extensions/omni.isaac.lab/omni/isaac/lab/managers/observation_manager.py

```diff
@@ -335,10 +351,18 @@ def _prepare_terms(self):
                 # check noise settings
                 if not group_cfg.enable_corruption:
                     term_cfg.noise = None
+                # check group history params and override terms
+                if group_cfg.history_length is not None:
+                    term_cfg.history_length = group_cfg.history_length
+                    term_cfg.flatten_history_dim = group_cfg.flatten_history_dim
                 # add term config to list to list
                 self._group_obs_term_names[group_name].append(term_name)
                 self._group_obs_term_cfgs[group_name].append(term_cfg)
-
+                # create history buffers
+                if term_cfg.history_length > 0:
+                    group_entry_history_buffer[term_name] = CircularBuffer(
+                        max_len=term_cfg.history_length, batch_size=self._env.num_envs, device=self._env.device
+                    )
                 # call function the first time to fill up dimensions
                 obs_dims = tuple(term_cfg.func(self._env, **term_cfg.params).shape)
```

Comment:
```suggestion
                # call function the first time to fill up dimensions
                obs_dims = tuple(term_cfg.func(self._env, **term_cfg.params).shape)
                # create history buffers
                if term_cfg.history_length > 0:
                    group_entry_history_buffer[term_name] = CircularBuffer(
                        max_len=term_cfg.history_length, batch_size=self._env.num_envs, device=self._env.device
                    )
                    obs_dims = (term_cfg.history_length * obs_dims[0], *obs_dims[1:])
                    if term_cfg.flatten_history_dim:
                        obs_dims = (np.prod(obs_dims),)
```
This populates the correct `_group_obs_term_dim` into the dictionary. Later, this is used to compute the `_group_obs_dim` however, I assume that its computation does not need any changes.